src/server/queue_func.c

```diff
@@ -159,6 +160,15 @@ que_free(pbs_queue *pq)
 
 		pdef->at_free(pattr);
 	}
+	/* free default chunks set on queue */
+	pkvp = pq->qu_seldft;
+	if (pkvp) {
+		for (i=0; i < pq->qu_nseldft; ++i) {
+			free((pkvp+i)->kv_keyw);
```

Comment:
should we first check (pkvp+i) for not null?

Reply:
qu_nseldft is actually a count of how many select defaults are there in qu_seldft. This gets set in "deflt_chunk_action()" function. Adding an additional check is not a bad thing to have but practically these two numbers have to be same so the memory must exist (unless there is a bug in deflt_chunk_action function). 
Similar code exists in deflt_chunk_action function as well. I would like to leave it as is unless you feel strongly about adding a null check.

Reply:
POSIX defines free(NULL) as a noop.  There's no real need for the check.

Reply:
Thanks @arungrover for explanation. I think we don't need null check for (pkvp+i) as you said it much be valid memory as per "deflt_chunk_action()" function.

@bhroam Indeed free(NULL) is noop, but we were taking about check for "(pkvp+i)" not "(pkvp+i)->kv_keyw" as we are checking for "pkvp" as like 165 but we are not checking "(pkvp+i)" anywhere. Hope you understand 😄